### DISCUSSION TITLE
KEDA External Scaler on StatefulSet

### DISCUSSION THREAD

**Discussion by spalaka** [kedacore/keda#1242] [Q&A / Need Help] [answered]
Hi Experts,
I am developing an external scaler (https://keda.sh/docs/1.5/scalers/external/) to scale the K8S instance by using custom parameters.

In a scaled object I couldn't refer to StatefulSet, it's allowing only a deployment name as per the spec (error: is invalid: spec.scaleTargetRef.deploymentName: Required value)

Please help me with the possibilities of how can I scale the statefulset deployment with an external scaler

Thanks,
SKP

**Comment by zroubalik** [comment, marked as answer]
Hi @spalaka.

You should build your scaler (https://keda.sh/docs/2.0/scalers/external/) based upcoming KEDA v2, it is adding support for StatefulSet and other CRs.

**Reply by spalaka** [reply]
Thanks @zroubalik for the prompt response, I have started putting the code accordingly.
When you say upcoming v2, when this can be available for production deployment.
And also will this help to scale the high available nodes simultaneously instead of just scaling standalone nodes.

**Reply by zroubalik** [reply]
We have a Candidate Release out and 2.0.0 should be out in a matter of week.

> And also will this help to scale the high available nodes simultaneously instead of just scaling standalone nodes.

Not sure what are you asking for? KEDA doesn't support scaling of nodes or anything related
